components/org.wso2.carbon.identity.claim.verification.endpoint/src/main/java/org/wso2/carbon/identity/claim/verification/endpoint/impl/InitVerificationApiServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+                (String) IdentityUtil.threadLocalProperties.get().get(ClaimVerificationEndpointConstants
+                        .TENANT_NAME_FROM_CONTEXT);
+
+        User user = ClaimVerificationEndpointUtils.getUser(verificationInitiatingRequest.getUser(),
+                tenantDomainFromContext);
+
+        int tenantIdFromContext = IdentityTenantUtil.getTenantId(user.getTenantDomain());
+
+        String[] userList = ClaimVerificationEndpointUtils.getUserList(tenantIdFromContext, user.getUsername());
+
+        // Validate incoming user data.
+        if (ArrayUtils.isEmpty(userList)) {
+            String msg = "Unable to find an user with username: " + user.getUsername() + " in the system.";
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(msg);
+            }
+            ClaimVerificationEndpointUtils.handleBadRequest(
+                    ClaimVerificationEndpointConstants.ERROR_CODE_NO_MATCHING_USER_FOUND, msg);
+        } else if (userList.length == 1) {
+            user.setRealm(IdentityUtil.extractDomainFromName(userList[0]));
+        } else {
+            String msg = "There are multiple users with username: " + user.getUsername() + " in the system, " +
+                    "please send the correct user-store domain along with the username.";
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(msg);
+            }
+            ClaimVerificationEndpointUtils.handleBadRequest(
+                    ClaimVerificationEndpointConstants.ERROR_CODE_MULTIPLE_MATCHING_USERS_FOUND, msg);
+        }
+
+        Claim claim = ClaimVerificationEndpointUtils.getClaim(verificationInitiatingRequest.getClaim());
+        org.wso2.carbon.user.api.Claim claimMetaData =
+                ClaimVerificationEndpointUtils.getClaimMetaData(tenantIdFromContext, claim.getClaimUri());
+
+        // Validate incoming claim data.
+        if (claimMetaData == null) {
+            String msg = "Unable to find a claim with claim uri: " + claim.getClaimUri() + " in the system.";
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(msg);
+            }
+            ClaimVerificationEndpointUtils.handleBadRequest(
+                    ClaimVerificationEndpointConstants.ERROR_CODE_NO_MATCHING_USER_FOUND, msg);
+        }
+        claim.setClaimDisplayTag(claimMetaData.getDisplayTag());
+
+        String confirmationCode = "";
+        try {
+            confirmationCode = ClaimVerificationEndpointUtils.getClaimVerificationHandler().initVerification(
+                    user, claim, verificationInitiatingRequest.getVerificationMethod(),
+                    ClaimVerificationEndpointUtils.getPropertiesToMap(verificationInitiatingRequest.getProperties()));
+        } catch (ClaimVerificationException e) {
+
+            if (e instanceof ClaimVerificationBadRequestException) {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug(e.getErrorCode() + ":" + e.getMessage(), e);
```

Comment:
Shall we add some context to the debug log instead of just logging the error code and message? Do the same to other similar places.

Reply:
Added